apps/backoffice-v2/src/pages/Entity/components/Case/components/CaseOptions/options/EditCollectionFlow/hooks/useEditCollectionFlow/useEditCollectionFlow.ts

```diff
@@ -0,0 +1,44 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { useUpdateWorkflowByIdMutation } from '@/domains/workflows/hooks/mutations/useUpdateWorkflowByIdMutation/useUpdateWorkflowByIdMutation';
+import { updateStateForEditing } from './helpers/update-state-for-editing';
+import { useCallback } from 'react';
+import { toast } from 'sonner';
+import { useEditCaseStateMutation } from './hooks/useEditCaseStateMutation/useEditCaseStateMutation';
+
+interface IUseEditCollectionFlowProps {
+  workflowId: string;
+  workflowContext: TWorkflowById['context'];
+}
+
+export const useEditCollectionFlow = ({
+  workflowId,
+  workflowContext,
+}: IUseEditCollectionFlowProps) => {
+  const { mutate: updateWorkflowById, isLoading: isUpdatingWorkflow } =
+    useUpdateWorkflowByIdMutation({
+      workflowId,
+    });
+  const { mutateAsync: editCaseState, isLoading: isEditCaseStateLoading } =
+    useEditCaseStateMutation();
+
+  const onEditCollectionFlow = useCallback(async () => {
+    try {
+      const updatedWorkflowContext = updateStateForEditing(workflowContext);
+
+      // Updating case state first to avoid unnecessary context update in case this step fails
+      await editCaseState({ workflowId });
+      updateWorkflowById({
+        context: updatedWorkflowContext,
+        action: 'edit_collection_flow',
+      });
+    } catch (error) {
+      console.error(error);
+      toast.error('Collection flow missing state.');
```

Comment:
This copy should be both from the translations json and "user-facing". Something along the lines of how the agent's action failed, not the cause. In this case the agent won't be able to do anything about a missing state, nor he would know what a state is.